torch/utils/checkpoint.py

```diff
@@ -45,32 +70,42 @@ def check_backward_validity(inputs: Iterable[Any]) -> None:
 def get_device_states(*args) -> Tuple[List[int], List[torch.Tensor]]:
     # This will not error out if "arg" is a CPU tensor or a non-tensor type because
     # the conditionals short-circuit.
-    fwd_gpu_devices = list({arg.get_device() for arg in args
-                            if isinstance(arg, torch.Tensor) and arg.is_cuda})
+    fwd_device_ids = list({arg.get_device() for arg in args
+                          if isinstance(arg, torch.Tensor) and not arg.device.type == "cpu"})
+
+    fwd_device_states = []
+    device_module = _get_device_module(infer_device_type(*args))
 
-    fwd_gpu_states = []
-    for device in fwd_gpu_devices:
-        with torch.cuda.device(device):
-            fwd_gpu_states.append(torch.cuda.get_rng_state())
+    for device_id in fwd_device_ids:
+        with device_module.device(device_id):
+            fwd_device_states.append(device_module.get_rng_state())
 
-    return fwd_gpu_devices, fwd_gpu_states
+    return fwd_device_ids, fwd_device_states
 
 
 def set_device_states(devices, states) -> None:
+    device_module = _get_device_module(infer_device_type(states))
     for device, state in zip(devices, states):
-        with torch.cuda.device(device):
-            torch.cuda.set_rng_state(state)
+        with device_module.device(device):
+            device_module.set_rng_state(state)
 
-def _get_autocast_kwargs():
-    gpu_autocast_kwargs = {"enabled": torch.is_autocast_enabled(),
-                           "dtype": torch.get_autocast_gpu_dtype(),
-                           "cache_enabled": torch.is_autocast_cache_enabled()}
+
+def _get_autocast_kwargs(device="cuda"):
+    if device == "cuda":
+        device_autocast_kwargs = {"enabled": torch.is_autocast_enabled(),
+                                  "dtype": torch.get_autocast_gpu_dtype(),
+                                  "cache_enabled": torch.is_autocast_cache_enabled()}
+    else:
+        device_module = _get_device_module(device)
+        device_autocast_kwargs = {"enabled": device_module.is_autocast_enabled(),
+                                  "dtype": device_module.get_autocast_dtype(),
+                                  "cache_enabled": torch.is_autocast_cache_enabled()}
```

Comment:
Just to make sure, this is intentional right? (is there no such thing as `device_module.is_autocast_cache_enabled()`)